docs/multi_process.md

```diff
@@ -2,72 +2,111 @@
 
 ## Introduction
 
-This guide explains how to use JAX in environments such as [Cloud
-TPU](https://cloud.google.com/tpu) pods where accelerators are spread across
-multiple CPU hosts or JAX processes. We’ll refer to these as “multi-process”
-environments.
+This guide explains how to use JAX in environments such as
+GPU clusters and [Cloud TPU](https://cloud.google.com/tpu) pods where
+accelerators are spread across multiple CPU hosts or JAX processes. We’ll refer
+to these as “multi-process” environments.
 
 This guide specifically focuses on how to use collective communication
-operations (e.g. {func}`jax.lax.psum`) in multi-process settings, although other
-communication methods may be useful too depending on your use case (e.g. RPC,
-[mpi4jax](https://github.com/mpi4jax/mpi4jax)). If you’re not already familiar
-with JAX’s collective operations, we recommend starting with the
-{doc}`/jax-101/06-parallelism` notebook. An important requirement of multi-process
-environments in JAX is direct communication links between accelerators, e.g. the
-high-speed interconnects for Cloud TPUs or
-[NCCL](https://developer.nvidia.com/nccl) for GPUs. These links are what allow
-collective operations to run across multiple processes’ worth of accelerators.
-
+operations (e.g. {func}`jax.lax.psum` ) in multi-process settings, although
+other communication methods may be useful too depending on your use case (e.g.
+RPC, [mpi4jax](https://github.com/mpi4jax/mpi4jax)). If you’re not already
+familiar with JAX’s collective operations, we recommend starting with the
+{doc}`/jax-101/06-parallelism` notebook. An important requirement of
+multi-process environments in JAX is direct communication links between
+accelerators, e.g. the high-speed interconnects for Cloud TPUs or
+[NCCL](https://developer.nvidia.com/nccl) for GPUs. These links allow
+collective operations to run across multiple processes’ worth of accelerators
+with high performance.
 
 ## Multi-process programming model
 
 Key concepts:
-*   You must run at least one JAX process per host.
-*   Each process has a distinct set of _local_ devices it can address. The
-    _global_ devices are the set of all devices across all processes.
-*   Use standard JAX parallelism APIs like {func}`~jax.pmap` and
-    {func}`~jax.experimental.maps.xmap`. Each process “sees” _local_ input and
+
+  * You must run at least one JAX process per host.
+  * You must initialize the cluster with {func}`jax.distributed.initialize`.
```

Comment:
This is not currently the case on TPU? (As in, most people don't call `jax.distributed.initialize` on multi-process Cloud TPU, but maybe they should/could?)

Reply:
It's not technically mandatory, but it's strongly recommended for health-checking and for making distributed checkpointing of GDAs work.

Reply:
I was gonna add this to the cloud_tpu_init code, so it works automatically on Cloud TPU pods. But I guess we can keep it as-is for now and change it later when I actually do that. I'm a bit worried this will confuse existing Cloud TPU users, but they probably already read this doc.

Reply:
@jekbradbury +1
Maybe we should try to make that "no config" experience work across the board ... I believe @nvcastet is working on making this the case for GPU clusters too. I think this is ok for now, but I guess we need to think more about this.